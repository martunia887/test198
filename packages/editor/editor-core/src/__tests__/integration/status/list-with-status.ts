import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
} from '../_helpers';

// TODO: safari keys do not work after upgrade
BrowserTestCase(
  'status.ts: Insert status into list, continue, then click back on status',
  { skip: ['ie', 'safari'] },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);
    await browser.waitForSelector(editable);
    await browser.click(editable);

    await browser.type(editable, '* abc ');
    await quickInsert(browser, 'Status');
    await browser.waitForSelector(`[aria-label="Popup"] input`);
    await browser.type(`[aria-label="Popup"] input`, 'DONE');
    await browser.click(editable);
    await browser.waitForSelector(`[aria-label="Popup"] input`, {}, true); // removed.
    await browser.type(editable, 'def');

    // Click status, and wait for popup
    await browser.click('.statusView-content-wrap');
    await browser.waitForSelector(`[aria-label="Popup"] input`);
    await browser.type(`[aria-label="Popup"] input`, ' WAITING');
    await browser.click(editable);

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
