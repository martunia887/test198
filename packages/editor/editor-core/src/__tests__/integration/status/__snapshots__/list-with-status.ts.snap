// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status.ts: Insert status into list, continue, then click back on status 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "abc ",
                  "type": "text",
                },
                Object {
                  "attrs": Object {
                    "color": "neutral",
                    "localId": "abc-123",
                    "style": "",
                    "text": "DONE WAITING",
                  },
                  "type": "status",
                },
                Object {
                  "text": " def",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;
